### PR TITLE
Don't create a migration if not necessary

### PR DESCRIPTION
**Pull request by albandum**
## Description

Making sure the `npm run create-db-migration` script doesn't unnecessarily create a migration

### CODE REVIEW

File: front/create_db_migration_file.sh [hunk 1]
```diff
@@ -47,8 +47,15 @@ diff --unified=0 --color=always main_output.txt current_output.txt
...
 # Run diff and extract only SQL statements.
 echo "Running diff and extracting SQL statements..."
-echo "-- Migration created on $current_date" > diff_output.txt
-diff --unified=0 main_output.txt current_output.txt | awk '/^\+[^+]/ {print substr($0, 2)}' >> diff_output.txt
+diff_output=$(diff --unified=0 main_output.txt current_output.txt | awk '/^\+[^+]/ {print substr($0, 2)}')
+if [ -n "$diff_output" ]; then
+  echo "-- Migration created on $current_date" > diff_output.txt
+  echo "$diff_output" >> diff_output.txt
+else
+    echo "No migration necessary. Cleaning up..."
```
Comment (flvndvd) [inline front/create_db_migration_file.sh:55]: `nit` you can create a function to clean up.

**Review by flvndvd — APPROVED** [review]

File: front/create_db_migration_file.sh [hunk 2]
```diff
@@ -6,6 +6,15 @@ if [ "$NODE_ENV" == "production" ]; then
...
 fi
 
+cleanup() {
+    echo "Cleaning up temporary files..."
+    rm -f main_output.txt current_output.txt
+    exit 0
+}
+
+# CLEANUP TEMP FILES ON EXIT
+trap 'cleanup' SIGINT SIGTERM EXIT
```
Comment (flvndvd) [inline front/create_db_migration_file.sh:16]: ❤️

**Review by flvndvd — APPROVED** [review]
🙏

**Comment by spolu** [comment]
Bash ninja

**Pull request merged**